### PR TITLE
Promote scheduled trial-clock smoke coverage

### PR DESCRIPTION
**Pull request by The-Obstacle-Is-The-Way**
## Summary

Promotes dev → main carrying PR #798 (`9443f321`): DEBT-468 Part 1 item 1 — trial day-8 behavior gains a recurring automated verification cadence. A hardened runner (`scripts/run-trial-clock-smoke.ts`, 37-case contract suite) fail-closes on missing/dummy credentials, executes the existing gated Stripe test-clock smoke through the integration config, and rejects skipped/partial/duplicate/malformed/non-passing results — a `describe.skip` outcome can never report success. `.github/workflows/stripe-trial-clock-smoke.yml` runs Sundays 07:17 UTC + manual dispatch, outside PR CI, zero new owner-provisioned secrets, credentials exposed only to the smoke step. Live test-mode proof: both cases passed in 19.9 s with zero-residual before/after cleanup censuses. The smoke's case bodies are untouched and remain skip-gated in the normal integration lane.

**Activation note:** GitHub registers scheduled/manual triggers only once the workflow exists on the default branch — merging this promotion is what turns the schedule on. The first hosted run (manual dispatch recommended) is the remaining verification receipt.

## Verification

Red/green contract receipts for every preflight and process branch; full local gate green (unit 3,914 / browser 398 / integration 256 + 2 smoke skips / build / E2E with trial-start passing); exact-head CI, Codecov, Vercel green; CodeRabbit approved after seven true-positive fixes and one receipted rebuttal; zero unresolved threads.

## Summary by CodeRabbit

- **New Features**
  - Added automated Stripe trial-clock smoke testing on a weekly schedule, with support for manual runs.
  - Added validation for expected trial scenarios, configuration, timeouts, interruptions, and incomplete results.
  - Added local safeguards to detect exhausted checkout replay capacity before end-to-end trial tests run.

- **Documentation**
  - Updated testing coverage and technical debt records, including replay-capacity limitations and follow-up work.

- **Tests**
  - Added comprehensive coverage for smoke-test execution, reporting, cleanup, failure handling, and replay-capacity enforcement.

### CODE REVIEW

**Comment by The-Obstacle-Is-The-Way** [comment]
@coderabbitai review

File: scripts/run-trial-clock-smoke.test.ts [hunk 1]
```diff
@@ -0,0 +1,628 @@
...
+    await expect(
+      spawnVitest({
+        command: process.execPath,
+        args: ['-e', "process.kill(process.pid, 'SIGTERM')"],
+        env: childEnvironment,
+      }),
+    ).rejects.toThrow(
+      'TRIAL_CLOCK_SMOKE_PROCESS_FAILED: Vitest ended with signal SIGTERM',
+    );
+  });
```
[the remark this answers is not in the document]
Comment (The-Obstacle-Is-The-Way) [reply scripts/run-trial-clock-smoke.test.ts]: Fixed in ce609557: both cases now pass an explicit EventEmitter (and 5s timeout), so no handlers ever attach to the worker process. The 50ms-sleep case was also rebuilt with a ready-file handshake and 5s budget, plus a new contract test proving SIGKILL escalation when a child ignores forwarded signals.

File: scripts/run-trial-clock-smoke.test.ts [hunk 2]
```diff
@@ -0,0 +1,628 @@
...
+      await new Promise((resolve) => setTimeout(resolve, 50));
+
+      parentSignals.emit(signal);
+
+      await expect(run).rejects.toThrow(
+        `TRIAL_CLOCK_SMOKE_PROCESS_FAILED: Vitest ended with signal ${signal}`,
+      );
+      expectNoParentSignalListeners(parentSignals);
+    },
+  );
```
[the remark this answers is not in the document]
Comment (The-Obstacle-Is-The-Way) [reply scripts/run-trial-clock-smoke.test.ts]: Fixed in ce609557: both cases now pass an explicit EventEmitter (and 5s timeout), so no handlers ever attach to the worker process. The 50ms-sleep case was also rebuilt with a ready-file handshake and 5s budget, plus a new contract test proving SIGKILL escalation when a child ignores forwarded signals.

File: scripts/run-trial-clock-smoke.ts [hunk 3]
```diff
@@ -0,0 +1,380 @@
...
+      `PROOF_NONPASSING_CASE: smoke report contains ${nonPassing.length} non-passing cases`,
+    );
+  }
+
+  return {
+    executed: assertions.length,
+    passed: assertions.length,
+    skipped: 0,
+  };
+}
```
[the remark this answers is not in the document]
Comment (The-Obstacle-Is-The-Way) [reply scripts/run-trial-clock-smoke.ts]: Fixed in ce609557: three fixture contract tests now feed real captured Vitest 4.1.10 reporter payloads through assertSmokeExecuted — a genuine skip-shaped capture (rejected), a genuine mixed-failure capture from a live provider timeout (rejected via success:false), and the captured envelope normalized to passed (accepted). Fixtures sanitized: paths and provider error text redacted.

File: scripts/run-trial-clock-smoke.ts [hunk 4]
```diff
@@ -0,0 +1,380 @@
...
+    return;
+  }
+
+  try {
+    killProcessGroup(-child.pid, signal);
+  } catch {
+    // The group may already have exited between the timeout and the signal.
+    child.kill(signal);
+  }
+}
```
[the remark this answers is not in the document]
Comment (The-Obstacle-Is-The-Way) [reply scripts/run-trial-clock-smoke.ts]: Addressed in ce609557; see the runner-hardening commit for the receipt.

File: scripts/run-trial-clock-smoke.ts [hunk 5]
```diff
@@ -0,0 +1,380 @@
...
+            : `signal ${signal}`;
+        reject(
+          new Error(
+            `TRIAL_CLOCK_SMOKE_PROCESS_FAILED: Vitest ended with ${reason}`,
+          ),
+        );
+      });
+    });
+  });
+}
```
[the remark this answers is not in the document]
Comment (The-Obstacle-Is-The-Way) [reply scripts/run-trial-clock-smoke.ts]: Addressed in ce609557; see the runner-hardening commit for the receipt.

File: docs/debt/index.md [hunk 6]
```diff
@@ -1,9 +1,13 @@
...
...(or direct Windows child) to five minutes, forwards parent `SIGINT`/`SIGTERM`, and rejects skip-green, partial, missing, malformed, duplicate, or non-passing reports. Its 37 contract cases cover preflight, exact invocation, child-process start/exit/signal/timeout, process-tree/cancellation outcomes, scratch-report cleanup, and execution proof. The dedicated workflow is configured for Sundays at 07:17 UTC plus manual dispatch, scopes the repository's existing Stripe test key and monthly price secrets to its smoke step, and needs neither Postgres nor new owner-provisioned secrets. GitHub accepts those triggers after the workflow reaches this repository's default branch (`main`). A local test-mode proof passed both no-card-cancel and card-active cases in 19.9 seconds; fixed-label counts were 0 customers / 0 clocks before and after cleanup. The normal integration lane remains skip-gated, and the first GitHub-hosted run after default-branch activation remains an open verification receipt. Part 1 item 4, Parts 2–4, and DEBT-469-coordinated splitting remain open. The Active register is DEBT-337, DEBT-414, DEBT-464, DEBT-465, DEBT-466, DEBT-468, and DEBT-469. Next Debt ID remains DEBT-470.
```
[the remark this answers is not in the document]
Comment (The-Obstacle-Is-The-Way) [reply docs/debt/index.md]: Rebuttal with precedent: the register's convention preserves each dated stanza's as-of state (Active list and Next Debt ID are historical values, deliberately not harmonized), and same-day pairs use Latest + Later — see the four 2026-08-14 stanzas (Later/Later/Earlier/Earliest) and the 2026-08-13 duplicate Earliers. This exact class was rebutted and accepted on PR #798 (historical 'Later 2026-08-16' stanza) and the closer audit verified the chain conventions. Line 8 records the state at the trial-clock filing, before DEBT-470 existed; line 6 is current truth. Renaming would break the established label grammar without adding information.

File: tests/e2e/helpers/trial-checkout-replay-capacity.test.ts [hunk 7]
```diff
@@ -0,0 +1,179 @@
...
+    const services = createServices(sessions);
+
+    await expect(
+      assertLocalTrialCheckoutReplayCapacity({
+        env: createEnv(),
+        nowMs: () => NOW_MS,
+        services,
+      }),
+    ).resolves.toBeUndefined();
+  });
```
[the remark this answers is not in the document]
Comment (The-Obstacle-Is-The-Way) [reply tests/e2e/helpers/trial-checkout-replay-capacity.test.ts]: Fixed in 51032868: the countable fixture set now sits exactly at SUBSCRIPTION_CHECKOUT_REPLAY_TRAVERSAL_LIMIT (nine complete + one expired), so wrongly counting any of the ignored sessions (open, setup-mode, 24h-old, foreign-user, annual, standard-variant) crosses the boundary and rejects — a filter regression is now observable. Good catch.

**Comment by The-Obstacle-Is-The-Way** [comment]
@coderabbitai resolve
